### PR TITLE
Add a Junit rule to run test using the default aws profile

### PR DESCRIPTION
**Pull request by abrooksv**
## Motivation and Context
Replace #2193

Removes the limit that tests need to finish in 1 hour

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt [hunk 1]
```diff
@@ -64,7 +63,6 @@ class MockClientManager : AwsClientManager() {
          */
         fun useRealImplementations(disposable: Disposable) {
             val clientManager = AwsClientManager()
-            Disposer.register(disposable, clientManager)
```
Comment (rli) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt:67]: do these get disposed automatically?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    // ProfileCredentialProviderFactory is designed to be a singleton, so replace it with a new instance
+                    ExtensionTestUtil.maskExtensions(DefaultCredentialManager.EP_NAME, listOf(ProfileCredentialProviderFactory()), disposable)
+
+                    ApplicationManager.getApplication().replaceService(CredentialManager::class.java, DefaultCredentialManager(), disposable)
+                    ApplicationManager.getApplication().replaceService(AwsResourceCache::class.java, DefaultAwsResourceCache(), disposable)
+                    project.replaceService(AwsConnectionManager::class.java, DefaultAwsConnectionManager(project), disposable)
+
+                    val credentialIdentifier = CredentialManager.getInstance().getCredentialIdentifierById(DEFAULT_PROFILE_ID)
+                        ?: throw IllegalStateException("RunWithRealCredentials requires a default AWS profile!")
+                    val region = AwsRegionProvider.getInstance()["us-west-2"] ?: throw IllegalStateException("Can't locate us-west-2")
```
Comment (rli) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt:70]: probably should be:

`getEnv("AWS_DEFAULT_REGION") ?: us-west-2 ?: throw`

**Review by rli — APPROVED** [review]

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt:67]: Yes, I dove into the code and confirmed it.

https://github.com/JetBrains/intellij-community/blob/210998614a279c714bdc1652604bad5bfb3237e4/platform/service-container/src/com/intellij/serviceContainer/BaseComponentAdapter.kt#L158

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt:70]: Ah, fair

Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt:70]: Fixed

**Pull request merged**